Meadowlark-js/src/repository/MongoEntityRepository.ts

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+import { Collection, Db, MongoClient, WithId } from 'mongodb';
+import { ForeignKeyItem } from '../model/ForeignKeyItem';
+import { EntityInfo } from '../model/EntityInfo';
+import { Security } from '../model/Security';
+import { ValidationOptions } from '../model/ValidationOptions';
+import { DeleteResult, GetResult, OwnershipResult, PutResult } from './BaseDynamoRepository';
+
+interface Entity {
```

Comment:
I wonder if we could use this exact same structure on PostgreSQL, which has a JSON data type? Keep the single-table design so that we don't have to generate schema-code.

But that doesn't help us get foreign keys, which would be the primary reason to consider such an idea. Secondary reason would be to give people SQL instead of NoSQL.

Just thinking "out loud" here.